File: app/src/main/java/ru/timcock/mayday/data/RequestsRealisation.java
package ru.timcock.mayday.data;

import android.content.Context;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import ru.timcock.mayday.data.requests.*;

import java.lang.reflect.Array;
import java.util.ArrayList;

class RequestsRealisation {
    private void auth(AuthRequest req, Context context) {
        RequestBuilder.buildService()
                .auth(req)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void registrate(RegistrateRequest req, Context context) {
        RequestBuilder.buildService()
                .registrate(req)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void addDream(AddDreamRequest addDreamRequest, Context context) {
        RequestBuilder.buildService()
                .addDream(addDreamRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void deleteDream(DeleteSomeRequest deleteDreamRequest, Context context) {
        RequestBuilder.buildService()
                .deleteDream(deleteDreamRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void getDreams(GetSomeRequest getSomeRequest, Context context) {
        RequestBuilder.buildService()
                .getDreams(getSomeRequest)
                .enqueue(new Callback<ArrayList<Dream>>() {
                    @Override
                    public void onResponse(Call<ArrayList<Dream>> call, Response<ArrayList<Dream>> response) {

                    }

                    @Override
                    public void onFailure(Call<ArrayList<Dream>> call, Throwable t) {

                    }
                });
    }

    private void addGoal(AddGoalRequest addGoalRequest, Context context) {
        RequestBuilder.buildService()
                .addGoal(addGoalRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void deleteGoal(DeleteSomeRequest deleGoalRequest, Context context) {
        RequestBuilder.buildService()
                .deleteGoal(deleGoalRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void getGoals(GetSomeRequest getSomeRequest, Context context) {
        RequestBuilder.buildService()
                .getGoals(getSomeRequest)
                .enqueue(new Callback<ArrayList<Goal>>() {
                    @Override
                    public void onResponse(Call<ArrayList<Goal>> call, Response<ArrayList<Goal>> response) {

                    }

                    @Override
                    public void onFailure(Call<ArrayList<Goal>> call, Throwable t) {

                    }
                });
    }

    private void getSovet(GetSomeRequest getSomeRequest, Context context) {
        RequestBuilder.buildService()
                .getSovets(getSomeRequest)
                .enqueue(new Callback<ArrayList<Sovet>>() {
                    @Override
                    public void onResponse(Call<ArrayList<Sovet>> call, Response<ArrayList<Sovet>> response) {

                    }

                    @Override
                    public void onFailure(Call<ArrayList<Sovet>> call, Throwable t) {

                    }
                });
    }

    private void addNote(AddNoteRequest addNoteRequest, Context context) {
        RequestBuilder.buildService()
                .addNote(addNoteRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void deleteNote(DeleteSomeRequest deleNoteRequest, Context context) {
        RequestBuilder.buildService()
                .deleteNote(deleNoteRequest)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {

                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {

                    }
                });
    }

    private void getNotes(GetSomeRequest getSomeRequest, Context context) {
        RequestBuilder.buildService()
                .getNotes(getSomeRequest)
                .enqueue(new Callback<ArrayList<Note>>() {
                    @Override
                    public void onResponse(Call<ArrayList<Note>> call, Response<ArrayList<Note>> response) {

                    }

                    @Override
                    public void onFailure(Call<ArrayList<Note>> call, Throwable t) {

                    }
                });
    }
}

File: app/src/main/java/ru/timcock/mayday/data/RequestsRealisation.kt
package ru.timcock.mayday.data

import android.content.Context
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import ru.timcock.mayday.data.requests.*
import java.util.ArrayList

class RequestsRealisation {
    companion object {
        private fun auth(req: AuthRequest, context: Context) {
            RequestBuilder.buildService()
                .auth(req)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }


        private fun registrate(req: RegistrateRequest, context: Context) {
            RequestBuilder.buildService()
                .registrate(req)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun addDream(addDreamRequest: AddDreamRequest, context: Context) {
            RequestBuilder.buildService()
                .addDream(addDreamRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun deleteDream(deleteDreamRequest: DeleteDreamRequest, context: Context) {
            RequestBuilder.buildService()
                .deleteDream(deleteDreamRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun getDreams(getSomeRequest: GetSomeRequest, context: Context) {
            RequestBuilder.buildService()
                .getDreams(getSomeRequest)
                .enqueue(object: Callback<ArrayList<Dream>> {
                    override fun onResponse(call: Call<ArrayList<Dream>>, response: Response<ArrayList<Dream>>) {

                    }

                    override fun onFailure(call: Call<ArrayList<Dream>>, t: Throwable) {

                    }
                })
        }

        private fun addGoal(addGoalRequest: AddGoalRequest, context: Context) {
            RequestBuilder.buildService()
                .addGoal(addGoalRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun deleteGoal(deleGoalRequest: DeleteGoalRequest, context: Context) {
            RequestBuilder.buildService()
                .deleteGoal(deleGoalRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun getGoals(getSomeRequest: GetSomeRequest, context: Context) {
            RequestBuilder.buildService()
                .getGoals(getSomeRequest)
                .enqueue(object: Callback<ArrayList<Goal>> {
                    override fun onResponse(call: Call<ArrayList<Goal>>, response: Response<ArrayList<Goal>>) {

                    }

                    override fun onFailure(call: Call<ArrayList<Goal>>, t: Throwable) {

                    }
                })
        }

        private fun addSovet(addSovetRequest: AddSovetRequest, context: Context) {
            RequestBuilder.buildService()
                .addSovet(addSovetRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun deleteSovet(deleteSovetRequest: DeleteSovetRequest, context: Context) {
            RequestBuilder.buildService()
                .deleteSovet(deleteSovetRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun getSovet(getSomeRequest: GetSomeRequest, context: Context) {
            RequestBuilder.buildService()
                .getSovets(getSomeRequest)
                .enqueue(object: Callback<ArrayList<Sovet>> {
                    override fun onResponse(call: Call<ArrayList<Sovet>>, response: Response<ArrayList<Sovet>>) {

                    }

                    override fun onFailure(call: Call<ArrayList<Sovet>>, t: Throwable) {

                    }
                })
        }

        private fun addNote(addNoteRequest: AddNoteRequest, context: Context) {
            RequestBuilder.buildService()
                .addNote(addNoteRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun deleteNote(deleNoteRequest: DeleteNoteRequest, context: Context) {
            RequestBuilder.buildService()
                .deleteNote(deleNoteRequest)
                .enqueue(object: Callback<Void> {
                    override fun onResponse(call: Call<Void>, response: Response<Void>) {

                    }

                    override fun onFailure(call: Call<Void>, t: Throwable) {

                    }
                })
        }

        private fun getNotes(getSomeRequest: GetSomeRequest, context: Context) {
            RequestBuilder.buildService()
                .getNotes(getSomeRequest)
                .enqueue(object: Callback<ArrayList<Note>> {
                    override fun onResponse(call: Call<ArrayList<Note>>, response: Response<ArrayList<Note>>) {

                    }

                    override fun onFailure(call: Call<ArrayList<Note>>, t: Throwable) {

                    }
                })
        }
    }
}